internal/fshasher/fshasher.go

```diff
@@ -110,14 +111,21 @@ func header(ctx context.Context, fullpath string, e os.FileInfo) (*tar.Header, e
 }
 
 func writeDirectory(ctx context.Context, tw *tar.Writer, fullpath string, d fs.Directory) error {
-	err := d.IterateEntries(ctx, func(c context.Context, e fs.Entry) error {
-		if err2 := write(c, tw, path.Join(fullpath, e.Name()), e); err2 != nil {
-			return err2
-		}
+	all, err := fs.GetAllEntries(ctx, d)
```

Comment:
maybe write a single line comment explaining why we should not iterate so that people will not revert this again ?